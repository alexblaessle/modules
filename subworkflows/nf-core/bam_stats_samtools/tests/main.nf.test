nextflow_workflow {

    name "Test Workflow BAM_STATS_SAMTOOLS"
    script "../main.nf"
    workflow "BAM_STATS_SAMTOOLS"
    tag "samtools"
    tag "samtools/flagstat"
    tag "samtools/idxstats"
    tag "samtools/stats"
    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "bam_stats_samtools"

    test("test_bam_stats_samtools_single_end") {

        when {
            params {
                outdir   = "$outputDir"
            }
            workflow {
                """
                    input[0] = [ [ id:'test', single_end:true ], // meta map
                                file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam'], checkIfExists: true),
                                file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam_bai'], checkIfExists: true)
                        ]
                    input[1] = [ [ id:'genome' ],
                            file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                            ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("test_bam_stats_samtools_paired_end") {

        when {
            params {
                outdir   = "$outputDir"
            }
            workflow {
                """
                    input[0] = [ [ id:'test', single_end:true ], // meta map
                                file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                                file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true)
                        ]
                    input[1] = [ [ id:'genome' ],
                            file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                            ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("test_bam_stats_samtools_paired_end_cram") {

        when {
            params {
                outdir   = "$outputDir"
            }
            workflow {
                """
                     input[0] = [ [ id:'test', single_end:false ], // meta map
                                file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram'], checkIfExists: true),
                                file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_cram_crai'], checkIfExists: true)
                                ]
                    input[1] = [ [ id:'genome' ],
                            file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                            ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

}
