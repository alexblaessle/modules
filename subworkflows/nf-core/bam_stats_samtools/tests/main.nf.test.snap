{
    "test_bam_stats_samtools_single_end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.stats:md5,51bed22fd8a090041d9fee548e796db4"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.flagstat:md5,2191911d72575a2358b08b1df64ccb53"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.idxstats:md5,613e048487662c694aa4a2f73ca96a20"
                    ]
                ],
                "3": [
                    "versions.yml:md5,3bd690f1f37bd590d5a12cd3c01a6a97",
                    "versions.yml:md5,c149c00245809be026fcc77355bbe2ad",
                    "versions.yml:md5,ce498ec7965c8bce01c822169d083ecf"
                ],
                "flagstat": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.flagstat:md5,2191911d72575a2358b08b1df64ccb53"
                    ]
                ],
                "idxstats": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.idxstats:md5,613e048487662c694aa4a2f73ca96a20"
                    ]
                ],
                "stats": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.stats:md5,51bed22fd8a090041d9fee548e796db4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3bd690f1f37bd590d5a12cd3c01a6a97",
                    "versions.yml:md5,c149c00245809be026fcc77355bbe2ad",
                    "versions.yml:md5,ce498ec7965c8bce01c822169d083ecf"
                ]
            }
        ],
        "timestamp": "2023-10-18T09:18:05.5701316"
    },
    "test_bam_stats_samtools_paired_end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.stats:md5,307e96e39c8034bf39b3dc272ac543a4"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.flagstat:md5,4f7ffd1e6a5e85524d443209ac97d783"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.idxstats:md5,df60a8c8d6621100d05178c93fb053a2"
                    ]
                ],
                "3": [
                    "versions.yml:md5,3bd690f1f37bd590d5a12cd3c01a6a97",
                    "versions.yml:md5,c149c00245809be026fcc77355bbe2ad",
                    "versions.yml:md5,ce498ec7965c8bce01c822169d083ecf"
                ],
                "flagstat": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.flagstat:md5,4f7ffd1e6a5e85524d443209ac97d783"
                    ]
                ],
                "idxstats": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.idxstats:md5,df60a8c8d6621100d05178c93fb053a2"
                    ]
                ],
                "stats": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.stats:md5,307e96e39c8034bf39b3dc272ac543a4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3bd690f1f37bd590d5a12cd3c01a6a97",
                    "versions.yml:md5,c149c00245809be026fcc77355bbe2ad",
                    "versions.yml:md5,ce498ec7965c8bce01c822169d083ecf"
                ]
            }
        ],
        "timestamp": "2023-10-18T09:18:15.816429014"
    },
    "test_bam_stats_samtools_paired_end_cram": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.stats:md5,33051d4e251dad3da1db2d4d0ed0b1d1"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.flagstat:md5,a53f3d26e2e9851f7d528442bbfe9781"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.idxstats:md5,e179601fa7b8ebce81ac3765206f6c15"
                    ]
                ],
                "3": [
                    "versions.yml:md5,3bd690f1f37bd590d5a12cd3c01a6a97",
                    "versions.yml:md5,c149c00245809be026fcc77355bbe2ad",
                    "versions.yml:md5,ce498ec7965c8bce01c822169d083ecf"
                ],
                "flagstat": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.flagstat:md5,a53f3d26e2e9851f7d528442bbfe9781"
                    ]
                ],
                "idxstats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.idxstats:md5,e179601fa7b8ebce81ac3765206f6c15"
                    ]
                ],
                "stats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.stats:md5,33051d4e251dad3da1db2d4d0ed0b1d1"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3bd690f1f37bd590d5a12cd3c01a6a97",
                    "versions.yml:md5,c149c00245809be026fcc77355bbe2ad",
                    "versions.yml:md5,ce498ec7965c8bce01c822169d083ecf"
                ]
            }
        ],
        "timestamp": "2023-10-18T09:18:27.289474576"
    }
}